packages/expo-image-manipulator/ios/ImageManipulations.swift

```diff
@@ -0,0 +1,211 @@
+      context.concatenate(transform)
+    }
+    imageView.layer.render(in: context)
+  }
+}
+
+/**
+ Creates a new image by cropping given image to specified width and height.
+ */
+internal func manipulate(image: UIImage, crop: CropRect) throws -> UIImage {
+  let rect = crop.toRect()
+  let isOutOfBounds = rect.origin.x > image.size.width
+  || rect.origin.y > image.size.height
+  || rect.width > image.size.width
+  || rect.height > image.size.height
+
+  guard !isOutOfBounds else {
+    throw ImageInvalidCropError()
+  }
+  guard let cgImage = image.cgImage?.cropping(to: rect) else {
+    throw ImageCropFailedError(rect: rect)
+  }
+  return UIImage(cgImage: cgImage, scale: image.scale, orientation: image.imageOrientation)
+}
+
+/**
+ Makes sure the image is oriented up and not mirrored.
+ Guarantees that the original pixel data matches the displayed orientation.
+ */
+internal func fixImageOrientation(_ image: UIImage) throws -> UIImage {
+  guard let cgImage = image.cgImage else {
+    throw ImageNotFoundError()
+  }
+  guard let colorSpace = cgImage.colorSpace else {
+    // That should never happen as `colorSpace` is empty only when the image is a mask.
+    throw ImageColorSpaceNotFoundError()
+  }
+
+  var transform = CGAffineTransform.identity
+
+  switch image.imageOrientation {
+  case .down, .downMirrored:
+    transform = transform.translatedBy(x: image.size.width, y: image.size.height)
+    transform = transform.rotated(by: Double.pi)
+  case .left, .leftMirrored:
+    transform = transform.translatedBy(x: image.size.width, y: 0)
+    transform = transform.rotated(by: Double.pi / 2)
+  case .right, .rightMirrored:
+    transform = transform.translatedBy(x: 0, y: image.size.height)
+    transform = transform.rotated(by: -Double.pi / 2)
+  default:
+    break
+  }
+
+  switch image.imageOrientation {
+  case .upMirrored, .downMirrored:
+    transform = transform.translatedBy(x: image.size.width, y: 0)
+    transform = transform.scaledBy(x: -1, y: 1)
+  case .leftMirrored, .rightMirrored:
+    transform = transform.translatedBy(x: image.size.height, y: 0)
+    transform = transform.scaledBy(x: -1, y: 1)
+  default:
+    break
+  }
+
+  let context = CGContext(
+    data: nil,
+    width: Int(image.size.width),
+    height: Int(image.size.height),
+    bitsPerComponent: cgImage.bitsPerComponent,
+    bytesPerRow: 0,
+    space: colorSpace,
+    bitmapInfo: CGImageAlphaInfo.premultipliedLast.rawValue
+  )
+
+  guard let context = context else {
+    throw ImageContextLostError()
+  }
+
+  context.concatenate(transform)
+
+  switch image.imageOrientation {
+  case .left, .leftMirrored, .right, .rightMirrored:
+    context.draw(cgImage, in: CGRect(x: 0, y: 0, width: image.size.height, height: image.size.width))
+  default:
+    context.draw(cgImage, in: CGRect(x: 0, y: 0, width: image.size.width, height: image.size.height))
+    break
```

Comment:
**⚠️  SwiftLint Violation: Unneeded Break in Switch**
Avoid using unneeded break statements.